src/components/shared/Autosave.tsx

```diff
@@ -0,0 +1,50 @@
+import { debounce } from 'lodash';
+import { FC, memo, useCallback } from 'react';
+
+import { FieldValues, SubmitHandler, useFormContext, useWatch } from 'react-hook-form';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+
+interface IAutosaveProps {
+  onSubmit: SubmitHandler<FieldValues>;
+  isBlurred?: boolean;
+}
+
+/**
+ * Autosave component that needs to be wrapped inside a FormProvider. It will watch the form values and
+ * trigger the onSubmit function related to the form with a 3s debounce delay.
+ *
+ * @param onSubmit react-hook-forms onSubmit function
+ */
+const Autosave: FC<IAutosaveProps> = ({ onSubmit, isBlurred }) => {
```

Comment:
This component is not used any where, do we need it?

Reply:
Fixed